File: Software Development Module/Week7/EncryptApp.java
/*
EncryptApp.java
15/07/2020
Billy Cussen
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class EncryptApp {
    public static void main(String[]args){

        //Variables
        String input = new String();
        String output;
        boolean inputCheckEncrypt;

        //Object
        Scanner keyboard;
        Encrypt myEncryption;

        //Input;
        do{
            inputCheckEncrypt = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter a sentance: ");
                input = keyboard.nextLine();

                if(input.length() < 5){
                    throw new InputMismatchException("Please enter an input with a length of longer than 5");
                }

            }catch(InputMismatchException e){
                System.out.println(e.getMessage());
                inputCheckEncrypt = false;
            } finally{
                keyboard = null;
            }

        }while(!inputCheckEncrypt);

        myEncryption = new Encrypt();
        myEncryption.setInput(input);

        //Compute
        myEncryption.encryptInput();

        //Ouput
        output = myEncryption.getOutput();
        myEncryption = null;
        System.out.println(output);

    }
}

File: Python Course/Software Development Module - in Python/Week8/EncryptApp.py
"""
EncryptApp.py
Billy Cussen
29/09/2020
"""

#Objects
import gc
import Encrypt
from Encrypt import myEncryption

#Variables
repeat = "y"

#Greeting
print("***ENCRYPTION APP***")

while repeat == "y":

    inputCheck = True

    #User Input
    while inputCheck:
        try:
            sentance = str(input("Please enter the Sentance you want Encrypted: "))
        except:
            print("You need to enter a String (i.e. a Sentance or some characters..)")
        else: 
            if len(sentance) <= 5:
                print("You need to enter a String with a length of 5 Characters or more")
            else:
                inputCheck = False

    #Process and Output
    myDecryptionApp = myEncryption(sentance)
    myDecryptionApp.encryptSentance()
    print(myDecryptionApp.result)

    myDecryptionApp = None
    gc.collect()

    repeat = ""

    #Input - Retry?
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter if you would like to repeat(\"Y\" or \"N\"): "))
        repeat = repeat.lower()

        if repeat != "y" and repeat != "n":
            print ("You need to enter \"Y\" or \"N\"!")

print ("Thanks for using my App!")